plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/wizards/MissingFBTypeWizard.java

```diff
@@ -0,0 +1,147 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Sebastian Hollersbacher
+ *    - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.typemanagement.wizards;
+
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.fordiac.ide.model.commands.change.UpdateFBTypeCommand;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerFBNElement;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetworkElement;
+import org.eclipse.fordiac.ide.model.libraryElement.FBType;
+import org.eclipse.fordiac.ide.model.libraryElement.INamedElement;
+import org.eclipse.fordiac.ide.model.search.AbstractLiveSearchContext;
+import org.eclipse.fordiac.ide.model.search.types.FBInstanceSearch;
+import org.eclipse.fordiac.ide.model.typelibrary.FBTypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeEntry;
+import org.eclipse.fordiac.ide.model.ui.editors.DataTypeTreeSelectionDialog;
+import org.eclipse.fordiac.ide.model.ui.nat.FBTypeSelectionTreeContentProvider;
+import org.eclipse.fordiac.ide.model.ui.nat.TypeNode;
+import org.eclipse.fordiac.ide.ui.FordiacMessages;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+
+public class MissingFBTypeWizard extends AbstractMissingTypeWizard {
+	private static final String PAGE_NAME = "FBRepairPage"; //$NON-NLS-1$
+
+	public MissingFBTypeWizard(final ErrorMarkerFBNElement type) {
+		super(type);
+	}
+
+	@Override
+	protected AbstractRepairOperationPage createPage() {
+		return new FBRepairOperationPage(PAGE_NAME);
+	}
+
+	@Override
+	protected void handleChangeType() {
+		if (result instanceof final FBType fbtype && fbtype.getTypeEntry() instanceof final FBTypeEntry typeEntry) {
+			searchAndUpdate(typeEntry);
+		}
+	}
+
+	@Override
+	protected void handleCreateMissing() {
+		if (errorType instanceof final FBNetworkElement fbnElement) {
+			final TypeEntry typeEntry = RestrictedNewTypeWizard.showRestrictedNewTypeWizard(
+					fbnElement.getType().getName(), ".fbt", fbnElement.getTypeLibrary().getProject().getName()); //$NON-NLS-1$
+
+			if (typeEntry instanceof final FBTypeEntry fbtypeEntry) {
+				searchAndUpdate(fbtypeEntry);
+			}
+		}
+	}
+
+	private void searchAndUpdate(final FBTypeEntry newEntry) {
+		final FBInstanceSearch search = new FBInstanceSearch(((ErrorMarkerFBNElement) errorType).getType());
+		if (newEntry != null) {
+			final var res = search.performProjectSearch(newEntry.getTypeLibrary().getProject());
```

Comment:
wouldnt it be better to use a new **org.eclipse.fordiac.ide.model.search.types.SearchFilter** instead of searching the search result again ?

Reply:
Now that the BlockTypeInstanceSearch is used, this can not be changed nicely